Feature/arms

OliviaRS opened the pull request:
Most of this is Skyler's work (who isn't even on GitHub??) except for the last few commits, those are mine. Take a quick pass when you can. Thanks.

Commit:
Arms Subsystem

Added Arms subsytem and Armup and Armdown command created references for the right and left arms.

Commit:
Created gitignore files for any .o file

Commit:
Skyler's uncommitted work

Commit:
Added Arm On command

Removed unneeded commands

Commit:
Ready for PR

-fixed RobotMap
-naming
-organization

Commit:
Update ArmOn.cpp

duanem commented on Debug/src/.gitignore:
```diff
@@ -0,0 +1 @@
+*.o
```
We only need one git ignore file at the root directory. I'll bet smart git or eclipse is helping with these. 

You only need one.

duanem commented on src/Commands/ArmOn.cpp:
```diff
@@ -0,0 +1,41 @@
+
+ArmOn::ArmOn(double speed)
+{
+mspeed = speed;
```
Indent

duanem commented on src/Commands/ArmOn.cpp:
```diff
@@ -0,0 +1,41 @@
+mspeed = speed;
+	// Use Requires() here to declare subsystem dependencies
+	// eg. Requires(chassis);
+	Requires(Robot::arms.get());
```
Tabs to 2 spaces

duanem commented on src/OI.cpp:
```diff
@@ -31,6 +32,12 @@ OI::OI() {
     dB.reset(new JoystickButton(driver.get(), 2));
     dB->WhenReleased(new ToggleShifter());
 
+    dX.reset(new JoystickButton(driver.get(), 3));
```
We could use some symbolic names for these 2,3&4.

duanem commented on src/OI.cpp:
```diff
@@ -31,6 +32,12 @@ OI::OI() {
     dB->WhenReleased(new ToggleShifter());
 
+    dX.reset(new JoystickButton(driver.get(), 3));
+    dX->WhenReleased(new ArmOn(1.0));
```
WhenReleased doesn't seem right. WhenPressed?

duanem reviewed (commented):
FTP

Finished this pass.

There's a few comments that should be fixed.

MajorDarius committed:
@MajorDarius

-Added Id's to the buttons in OI.h
-Set up joystick button actions without proper commands

Commit:
Added static const int's to all the buttons to match the values

Commit:
incomplete

will complete later

Commit:
tweaked

moved enum to arm subsystem

Commit:
fixed SetArmLevel

duanem commented:
This is probably good enough to commit. Tuning the PID can come after this.

LGTM

Please merge.

duane701 commented:
Closing in favor of #7.
